counsel.el

```diff
@@ -5096,6 +5100,44 @@ build environments.")
                      (or (buffer-file-name) default-directory)
                      ".git")))
 
+;; This is loosely based on the bash completion code
+(defun counsel--get-make-calls (root)
+  "Return a list of make calls for a given directory.
+
+We search the Makefile for a list of PHONY targets which are generally
+the top-level targets a make system provides."
+  (let ((default-directory root))
+    (mapcar
+     (lambda(target)
+       (format "cd %s; make %s %s" root counsel-compile-make-opts target))
+     (split-string
+      (shell-command-to-string
+       (concat "make -nqp |"
+               "grep -B 1 PHONY |"
+               "grep ':' |"
+               "cut -d ':' -f 1 |"
+               "sort"))
+      "\n"))))
+
+(defun counsel-compile-get-make-invocaton (&optional build-root)
+  "Have a look in the root directory for any build control files.
+
+The optional `BUILD-ROOT' is useful for other helpers that have found
+  sub-directories that builds may be invoked in."
+  (let* ((root (or build-root (counsel--find-project-root)))
+         (local-files (directory-files root))
+         (compiles))
+    (mapc
+     (lambda (file)
+       (let ((iter (cond
+                    ((equalp file "makefile")
```

Comment:
There are at least also `Makefile` and `GNUMakefile`.  I guess that part of the code should become hookable via regular expression `defcustom`s.